tdp/core/deployment/deployment_plan.py

```diff
@@ -164,8 +193,25 @@ def from_failed_deployment(dag, deployment_log):
             remaining_operations = original_operations
             remaining_operation_names = original_operation_names
 
-        deployment_args = dict(
+        new_deployment_log = DeploymentLog(
             targets=remaining_operation_names,
             deployment_type=DeploymentTypeEnum.RESUME,
+            state=DeploymentStateEnum.PLANNED,
         )
-        return DeploymentPlan(remaining_operations, deployment_args)
+        return DeploymentPlan(remaining_operations, new_deployment_log)
+
+    @staticmethod
+    def from_deployment_log(
+        collections: Collections, deployment_log: DeploymentLog
+    ) -> "DeploymentPlan":
+        """Generate a DeploymentPlan from a DeploymentLog."""
```

Comment:
I think the doc should say that only a "planned" state for `DeploymentLog` is accepted.
```suggestion
        """Generate a DeploymentPlan from a "planned" DeploymentLog."""
```

Reply:
Done